### PR TITLE
feat: add target to install XBlocks into LMS+Studio

### PR DESCRIPTION
**Pull request by stvstnfrd**
Story
=====
As an XBlock developer,
I want to easily install my XBlock in LMS+Studio,
so I can easily run my XBlock in LMS+Studio.

Testing
=======
- Run `make dev.xblocks.install` once.
  - See the error/instruction message.
  - Follow directions and clone an XBlock to `edx-platform/src`.
- Run `make dev.xblocks.install` again.
  - Confirm your XBlock listed in the output.
- In Studio, enable your XBlock [1].
  - Confirm you can now add your XBlock to a Unit.

References
==========
- [1] https://edx.readthedocs.io/projects/open-edx-building-and-running-a-course/en/latest/exercises_tools/enable_exercises_tools.html#enable-additional-exercises-and-tools

----

I've completed each of the following, or confirmed they do not apply to this PR:

- [ ] Tested on both Mac and Linux (if changed Makefile or shell scripts)
    - [X] Already tested on: Linux
- [ ] Made a plan to communicate any major developer interface changes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -630,3 +630,31 @@ build-courses: ## Build course and provision studio, and ecommerce with it.
 	$(WINPTY) bash ./course-generator/create-courses.sh --studio --ecommerce course-generator/tmp-config.json
 	rm course-generator/tmp-config.json
 
+XBLOCKS_HOME=$(abspath $(DEVSTACK_WORKSPACE)/edx-platform/src)
+XBLOCKS_NAMES=$(addprefix src/,$(notdir $(wildcard $(XBLOCKS_HOME)/*)))
+dev.xblocks.install: dev.up.without-deps.lms dev.up.without-deps.studio  ## Install XBlocks into LMS+Studio
```
Comment (sarina) [inline Makefile]: image

File: Makefile [hunk 2]
```diff
@@ -630,3 +630,31 @@ build-courses: ## Build course and provision studio, and ecommerce with it.
...
+ifeq ($(XBLOCKS_NAMES),)
```
Comment (sarina) [inline Makefile]: How do I get to this clause? It went straight to the `else` but I want the instructions to install an XBlock shown to me. I wonder if you should add a statement to the `else` along the lines of, "if you wish to install additional blocks, do this:"

File: Makefile [hunk 3]
```diff
@@ -630,3 +630,31 @@ build-courses: ## Build course and provision studio, and ecommerce with it.
...
+	docker-compose exec studio env /edx/app/edxapp/devstack.sh exec \
+	    pip install -e $(XBLOCKS_NAMES) \
+	;
+	@printf '\n\n\n\n'
+	@printf "Here are the currently installed XBlocks "
+	@echo "(installed packages with 'xblock' in the name):"
+	@docker-compose exec studio env /edx/app/edxapp/devstack.sh exec \
+		pip freeze 2>/dev/null \
+		| grep -i xblock \
+		| sed 's/.*egg=//; s/\&.*//' \
```
Comment (sarina) [inline Makefile:652]: This is a nice display 👍

File: Makefile [hunk 4]
```diff
@@ -630,3 +630,32 @@ build-courses: ## Build course and provision studio, and ecommerce with it.
 	$(WINPTY) bash ./course-generator/create-courses.sh --studio --ecommerce course-generator/tmp-config.json
 	rm course-generator/tmp-config.json
 
+XBLOCKS_HOME=$(abspath $(DEVSTACK_WORKSPACE)/edx-platform/src)
+XBLOCKS_NAMES=$(addprefix src/,$(notdir $(wildcard $(XBLOCKS_HOME)/*)))
+dev.xblocks.install:  ## Install XBlocks into LMS+Studio
+	@printf '\n\n\n\n'
```
Comment (stvstnfrd) [inline Makefile]: Always show the instructions, instead of only when none are found.

File: Makefile [hunk 5]
```diff
@@ -630,3 +630,32 @@ build-courses: ## Build course and provision studio, and ecommerce with it.
...
+	@echo "This is setup to automatically install any XBlocks found in $(XBLOCKS_HOME)"
+	@echo "You'll need to check-out any XBlock repositories there, like:"
+	@echo
+	@echo "    git clone git@github.com:edx/xblock-image-modal.git $(XBLOCKS_HOME)/xblock-image-modal"
+	@echo
+ifeq ($(XBLOCKS_NAMES),)
+	@echo "then try this command again."
+	@exit 1
```
Comment (stvstnfrd) [inline Makefile]: but still bail if none can be found.

**Review by qasimgulzar — APPROVED** [review]

**Pull request closed without merging**